fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb

```diff
@@ -34,7 +34,24 @@ def run!
       parse_global_options
       remove_global_options options, @args
 
+      xcode_outdated = false
       begin
+        unless FastlaneCore::Helper.xcode_at_least?(Fastlane::MINIMUM_XCODE_VERSION)
+          xcode_outdated = true
+        end
+      rescue
+        # We don't care about exceptions here
+        # We'll land here if the user doesn't have Xcode at all for example
+        # which is fine for someone who uses fastlane just for Android project
+        # What we *do* care about is when someone links an old version of Xcode
+      end
+
+      begin
+        if xcode_outdated
```

Comment:
You should implement this out of the `begin` block.
I think `UI.user_error!` raises exception and it would be caught [here](https://github.com/fastlane/fastlane/pull/9609/files#diff-afee814ded0e79042832f3de1d681a53R109). Is this right?